quicktest.go

```diff
@@ -0,0 +1,106 @@
+// Licensed under the MIT license, see LICENCE file for details.
+
+package quicktest
+
+import (
+	"fmt"
+	"testing"
+)
+
+// New returns a new quick test checker, that is used to call Assert and Check.
+// The checker is instantiated passing the *testing.T value for the current
+// test or subtest. For instance:
+//
+//     func TestFoo(t *testing.T) {
+//         t.Run("A=42", func(t *testing.T) {
+//             c := qt.New(t)
+//             c.Assert(a, qt.Equals, 42)
+//         })
+//     }
+//
+// The library provides the Equals, CmpEquals, DeepEquals, ErrorMatches,
+// PanicMatches, IsNil and Not checkers. More can be added by implementing the
+// Checker interface.
+func New(t T) *C {
+	return &C{
+		t: t,
+	}
+}
+
+// C is a quick test checker used to call Assert and Check.
+type C struct {
+	t T
```

Comment:
I wonder if this should be:

    type C struct {
        T
    }

then the user has access to all the embedded T methods too, so they won't
need to pass around the T value as well to be able to call Fatal etc.

Also, I'm not sure whether it's worth including Run in T. It would be nice
to be able to use quicktest in benchmarks too. Perhaps you could embed
testing.TB  but dynamically test whether the T value implements Run
if C.Run is called, and fail otherwise?

Reply:
This is a good idea, done.